admin_guide/encrypting_data.adoc

```diff
@@ -0,0 +1,291 @@
+          secret: c2VjcmV0IGlzIHNlY3VyZQ== <6>
+        - name: key2
+          secret: dGhpcyBpcyBwYXNzd29yZA==
+    - secretbox:
+        keys:
+        - name: key1
+          secret: YWJjZGVmZ2hpamtsbW5vcHFyc3R1dnd4eXoxMjM0NTY=
+    - aesgcm:
+        keys:
+        - name: key1
+          secret: c2VjcmV0IGlzIHNlY3VyZQ==
+        - name: key2
+          secret: dGhpcyBpcyBwYXNzd29yZA==
+    - identity: {}
+----
+<1> Each `resources` array item is a separate config and contains a complete configuration.
+<2> The `resources.resources` field is an array of Kubernetes resource names (*resource* or
+*resource.group*) that should be encrypted.
+<3> The `providers` array is an ordered xref:encrypting-data-providers[list of
+the possible encryption providers]. Only one provider type may be specified per entry (*identity* or
+*aescbc* may be provided, but not both in the same item).
+<4> The first provider in the list is used to encrypt resources going into storage.
+<5> Arbitrary name of the secret.
+<6> Base64 encoded random key. Different providers have different key lengths. See instructions
+below xref:encrypting-data-process[how to generate key].
+====
+
+When reading resources from storage each provider that matches the stored data attempts to decrypt
+the data in order. If no provider can read the stored data due to a mismatch in format or secret
+key, an error is returned which prevents clients from accessing that resource.
+
+[IMPORTANT]
+====
+If any resource is not readable via the encryption config (because keys were changed),
+the only recourse is to delete that key from the underlying etcd directly. Calls attempting to
+read that resource will fail until it is deleted or a valid decryption key is provided.
+====
+
+[[encrypting-data-providers]]
+=== Available Providers
+
+[cols="1,1,1,1,1,3"]
+|===
+| Name | Encryption | Strength | Speed | Key Length | Other Considerations
+
+| *identity*
+| None
+| N/A
+| N/A
+| N/A
+| Resources written as-is without encryption. When set as the first provider, the resource will be
+decrypted as new values are written.
+
+| *aescbc*
+| AES-CBC with PKCS#7 padding
+| Strongest
+| Fast
+| 32-byte
+| The recommended choice for encryption but may be slightly slower than *secretbox*.
+
+| *secretbox*
+| XSalsa20 and Poly1305
+| Strong
+| Faster
+| 32-byte
+| A newer standard and may not be considered acceptable in environments that require high levels of
+review.
+
+| *aesgcm*
+| AES-GCM with a random pass:[<abbr title="Initialization Vector">IV</abbr>]
+| Must be rotated every 200k writes
+| Fastest
+| 16, 24, or 32-byte
+| *Is not recommended* for use except when an automated key rotation scheme is implemented.
+
+|===
+
+Each provider supports multiple keys. The keys are tried in order for decryption, and if the
+provider is the first provider, the first key is used for encryption.
+
+[NOTE]
+====
+Kubernetes has no proper nonce generator and uses a random IV as nonce for AES-GCM. Since AES-GCM
```

Comment:
Will our users know what `nonce` and `IV` are? I cannot find other references to these terms in the docs. 
Should probably spell out  `initialization vector (IV)` on first reference